base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/ScriptGeneratorSingleton.java

```diff
@@ -611,12 +638,59 @@ public void reloadScriptDefinitions() {
 		scriptDefinitionLoader.reloadScriptDefinitions();
 	}
 	
+	/**
+	 * Get list of data files from the directory where data files are saved.
+	 * 
+	 * @return list of files
+	 * @throws FileNotFoundException when DataFiles folder does not exist
+	 */
+	public List<String> getListOfdataFiles() throws FileNotFoundException {
+		return this.scriptGenFileHandler.getListOfdataFiles();
+	}
+	
+	/**
+	 * Loads parameter values from a file.
+	 * 
+	 * @param fileName name of data file user wants to load
+	 * @throws NoScriptDefinitionSelectedException when script definition has not been selected
+	 * @throws ScriptDefinitionNotMatched when the script definition used to generate data file does not match with the one used to load it
+	 */
+	public void loadParameterValues(String fileName) throws NoScriptDefinitionSelectedException, ScriptDefinitionNotMatched, UnsupportedOperationException {
+		ScriptDefinitionWrapper scriptDefinition = getScriptDefinition()
+				.orElseThrow(() -> new NoScriptDefinitionSelectedException("No Configuration Selected"));
+		List<Map<ActionParameter, String>> list = scriptGenFileHandler.getParameterValues(preferenceSupplier.scriptGeneratorDataFileFolder() + fileName + ".json", 
```

Comment:
You're using `.json` all over the place can you pull it out into a constant

Reply:
Done.